src/main/java/com/linkedin/xinfra/monitor/services/OffsetCommitServiceFactory.java

```diff
@@ -74,9 +74,7 @@ private Properties prepareConfigs(Map<String, Object> props) {
 
     Map<String, String> customProps = (Map<String, String>) props.get(CommonServiceConfig.CONSUMER_PROPS_CONFIG);
     if (customProps != null) {
-      for (Map.Entry<String, String> entry : customProps.entrySet()) {
-        consumerProps.put(entry.getKey(), entry.getValue());
-      }
+      consumerProps.putAll(customProps);
```

Comment:
never would have spotted this myself - putAll is a convenient shortcut + also more performant (I know, not relevant here)